crates/typst/src/model/list.rs

```diff
@@ -176,38 +176,33 @@ fn layout_list(
     });
 
     let Depth(depth) = ListElem::depth_in(styles);
-    let marker = elem
-        .marker(styles)
-        .resolve(engine, styles, depth)?
-        // avoid '#set align' interference with the list
-        .aligned(HAlignment::Start + VAlignment::Top);
-
-    let mut cells = vec![];
-    let mut locator = locator.split();
+    let marker: Content = elem.marker(styles).resolve(engine, styles, depth)?;
 
-    for item in elem.children() {
-        cells.push(Cell::new(Content::empty(), locator.next(&())));
-        cells.push(Cell::new(marker.clone(), locator.next(&marker.span())));
-        cells.push(Cell::new(Content::empty(), locator.next(&())));
-        cells.push(Cell::new(
-            item.body.clone().styled(ListElem::set_depth(Depth(1))),
-            locator.next(&item.body.span()),
-        ));
+    let pad = body_indent + indent; // TODO: plus marker width
```

Comment:
I think i need to measure the width of marker here. But I haven't figure out how to do it.